cgi-bin/DW/Controller/Admin/SendMail.pm

```diff
@@ -41,96 +42,112 @@ sub index_controller {
-                LJ::send_mail( $send );
-                $sent = 1;
-            }
-        } else {
-            $send->{to} = $sendto;
-            LJ::send_mail( $send );
-            $sent = 1;
+        } else { # no $sendto
+            $errors->add( "sendto", ".error.norcpt" );
         }
 
-        return error_ml( '/admin/sendmail/index.tt.error.nouseremail' )
-            unless $sent;
+        # at this point, we should have good form data; the form can still
+        # have errors below, but they aren't the fault of the user input
+
+        unless ( $errors->exist ) {
+            # now that we have the data, send the message.
+            # 1. insert data into siteadmin_email_history table
+            my $msgid = LJ::alloc_global_counter('N');
+            my $dbh = LJ::get_db_writer();
+            $dbh->do( "INSERT INTO siteadmin_email_history (msgid, remoteid," .
+                      " time_sent, account, sendto, subject, request, message, " .
+                      " notes) VALUES (?,?,?,?,?,?,?,?,?)", undef,
+                      $msgid, $remote->id, time, $account, $sendto, $subject,
+                      $support_req, $message, $teamnotes )
+                or return error_ml( $scope->( '.error.sendfailed' ) );
+            # keeping this as error_ml; if we get a DB error things are FUBAR
+
+            # 2. construct the message and send it to the user(s)
+            # (this block adapted from bin/worker/paidstatus)
+            my $msg = { from => "$account\@$LJ::DOMAIN",
+                        fromname => $LJ::SITENAME,
+                        subject => $subject,
+                        body => $message,
+                      };
+            my $sent = 0;
+
+            my $send = sub { LJ::send_mail( $msg ); $sent = 1; };
+
+            if ( $u && $u->is_community ) {
+                # send an email to every maintainer
+                my $maintus = LJ::load_userids( $u->maintainer_userids );
+                foreach my $maintu ( values %$maintus ) {
+                    $send->() if $msg->{to} = $maintu->email_raw;
+                }
+
+            } elsif ( $u ) {
+                $send->() if $msg->{to} = $u->email_raw;
+
+            } else {
+                $send->() if $msg->{to} = $sendto;
+            }
 
-        # 3. update userlog and return success message
-        $remote->log_event( 'siteadmin_email', { account => $account, msgid => $msgid } );
-        return success_ml( '/admin/sendmail/index.tt.success.msgtext', undef,
-            [ { text => LJ::Lang::ml( '/admin/sendmail/index.tt.success.linktext.a' ),
-                url => '/admin/sendmail' } ] );
+            # 3. update userlog and return success message
+            if ( $sent ) {
+                $remote->log_event( 'siteadmin_email', { account => $account,
+                                                         msgid => $msgid } );
+                return success_ml( $scope->( '.success.msgtext' ), undef,
```

Comment:
If you use render_success instead, you can just pass in the scope as an argument. possibly easier?